### PR TITLE
[14.0] partner_helper: add strict mode that raise an error if address is too long

### PR DESCRIPTION
**Pull request by sebastienbeau**
Add an optional strict mode, so you can raise an error if the address can not be splitted correctly because it's too long

@bealdav @florian-dacosta

### CODE REVIEW

File: partner_helper/models/partner.py [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import models
+from odoo import _, models
+from odoo.exceptions import UserError
 
 
-def split_char(char, output_number, size):
-    words = char.split(" ")
+def split_char(char, output_number, size, strict):
```
Comment (florian-dacosta) [inline partner_helper/models/partner.py]: Isn't it better to set strict=False in signature in case some modules use this method directly ?

**Review by florian-dacosta — APPROVED** [review]
LGTM (minor  comment)
Please port this to last version of the module

File: partner_helper/models/partner.py (hunk 1 above)
Comment (bealdav) [reply partner_helper/models/partner.py]: yes, it avoid to break things

**Comment by bealdav** [comment]
The name of this module is now https://github.com/OCA/partner-contact/tree/16.0/partner_address_split

File: partner_helper/models/partner.py (hunk 1 above)
Comment (sebastienbeau) [reply partner_helper/models/partner.py]: done

**Review by bealdav — APPROVED** [review]

**Review by dreispt — APPROVED** [review]

**Pull request merged**